lib/Sema/ConstraintSystem.cpp

```diff
@@ -171,11 +171,11 @@ void ConstraintSystem::assignFixedType(TypeVariableType *typeVar, Type type,
         continue;
 
       auto anchor = locator->getAnchor();
-      if (!anchor)
+      if (!(anchor && anchor.is<const Expr *>()))
```

Comment:
`getAsExpr`?

Reply:
Sure! I'd have to add a null check to `getLiteralProtocol`.

Reply:
Oh I was just thinking of:

```
auto anchor = getAsExpr(locator->getAnchor())
if (!anchor)
  continue;
```

Reply:
Ah yeah, makes sense :)